### PR TITLE
Rebalance strategies

### PR DESCRIPTION
**Pull request by jonels-msft**

### CODE REVIEW

**Comment by jonels-msft** [comment]
Fixes #900

File: develop/api_metadata.rst [hunk 1]
```diff
@@ -346,6 +346,129 @@ The pg_dist_colocation table contains information about which tables' shards sho
...
+| node_capacity_function         |         regproc      | | Identifier for a capacity function, which must take a nodeid as int,    |
+|                                |                      | | and return its notion of node capacity as type real                     |
++--------------------------------+----------------------+---------------------------------------------------------------------------+
+| shard_allowed_on_node_function |         regproc      | | Identifier for a function that given shardid bigint, and nodeidarg int, |
+|                                |                      | | returns boolean for whether the shard is allowed to be stored on the    |
+|                                |                      | | node                                                                    |
++--------------------------------+----------------------+---------------------------------------------------------------------------+
+| default_threshold              |         float4       | | Replication factor for all tables in this co-location group.            |
```
[the remark this answers is not in the document]
Comment (jonels-msft) [reply develop/api_metadata.rst]: Oops, copy and paste error, didn't notice I hadn't changed that one.

**Comment by jonels-msft** [comment]
Thanks @JelteF -- I updated it with your feedback. All good now?

File: develop/api_udf.rst [hunk 2]
```diff
@@ -1121,6 +1183,63 @@ Example
...
+
+**shard_cost_function:** identifies the function used to determine the "cost" of each shard
+
+**node_capacity_function:** identifies the function to measure node capacity
+
+**shard_allowed_on_node_function:** identifies the function which determines which shards can be placed on which nodes
+
+**default_threshold:** a floating point threshold that tunes how precisely the cumulative shard cost should be balanced between nodes
+
+**minimum_threshold:** (Optional) a safeguard column that holds the minimum value allowed for the default_threshold. Its default value is 0
```
Comment (JelteF) [inline develop/api_udf.rst]: Same here, is minimum for the `threshold` argument not `default_threshold`.

**Pull request merged**